java-engine/src/test/java/io/getunleash/engine/YggdrasilFFITest.java

```diff
@@ -0,0 +1,117 @@
+package io.getunleash.engine;
+
+import com.sun.jna.Pointer;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.lang.ref.Cleaner;
+import java.lang.ref.PhantomReference;
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
+import java.lang.reflect.Field;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.verify;
+
+class YggdrasilFFITest {
+
+    static final String VALID_PATH = "../target/release";
+    @Test
+    void testSuccessfulLibraryLoad() {
+        YggdrasilFFI ffi = new YggdrasilFFI(VALID_PATH);
+        assertNotNull(ffi);
+    }
+
+    @Test
+    void testFailedLibraryLoad() {
+        assertThrows(UnsatisfiedLinkError.class, () ->
+            new YggdrasilFFI("/invalid/path")
+        );
+    }
+
+    @Test
+    void testEngineMethods() {
+        YggdrasilFFI ffi = new YggdrasilFFI(VALID_PATH);
+        Pointer state = ffi.takeState("someToggles");
+        assertNotNull(state);
+        ffi.freeResponse(state);
+    }
+
+    @Test
+    void testLibraryPathVariations() {
+        assertDoesNotThrow(() -> {
+            new YggdrasilFFI(absoluteValidPath());
+        });
+
+        assertThrows(UnsatisfiedLinkError.class, () -> {
+            new YggdrasilFFI("/non/existent/path");
+        });
+    }
+
+    @Test
+    void testResourceCleanup() {
```

Comment:
This test is important, for details about how/why this works, take a deep breath and look into https://inside.java/2022/05/27/testing-clean-cleaner-cleanup/